R/pwr_1var.R

```diff
@@ -1,639 +1,1033 @@
-    },
-
-    #### Init table ----
-    .initPowerTab = function(results) {
-      table <- self$jaspResults[["powertab"]]
-      if (is.null(table)) {
-        # Create table if it doesn't exist yet
-        table <- createJaspTable(title = gettext("A Priori Power Analysis"))
-        table$dependOn(c(
-          "test",
-          "es",
-          "power",
-          "n",
-          "alt",
-          "alpha",
-          "calc",
-          "n_ratio"
-        ))
-        table$position <- 2
-        self$jaspResults[["powertab"]] <- table
-      } else {
-        return()
-      }
-
-      calc <- self$options$calc
-
-      if (calc == "n") {
-        order <- c(1, 2, 3, 4)
-      } else if (calc == "es") {
-        order <- c(2, 1, 3, 4)
-      } else if (calc == "power") {
-        order <- c(3, 1, 2, 4)
-      } else {
-        order <- c(4, 1, 2, 3)
-      }
-
-      colNames <- c("n", "es", "power", "alpha")
-      colLabels <- c(
-        "N",
-        gettext("Effect Size"),
-        gettext("Power"),
-        "\u03B1"
-      )
-      colType <- c("integer", "number", "number", "number")
-
-      for (i in seq_along(order)) {
-        table$addColumnInfo(colNames[order[i]],
-                            title = colLabels[order[i]],
-                            overtitle = if (i > 1) gettext("User Defined") else NULL,
-                            type = colType[order[i]]
-        )
-      }
-
-      row <- list()
-      for (i in 2:4) {
-        row[[colNames[order[i]]]] <- self$options[[colNames[order[i]]]]
-      }
-
-      table$addRows(rowNames = 1, row)
-
-      private$.populatePowerTab(results)
-    },
-    .initPowerESTab = function(results, stats) {
-      table <- self$jaspResults[["powerEStab"]]
-      if (is.null(table)) {
-        # Create table if it doesn't exist yet
-        table <- createJaspTable(title = gettext("Power by Effect Size"))
-        table$dependOn(c(
-          "test",
-          "es",
-          "power",
-          "n",
-          "alt",
-          "alpha",
-          "calc",
-          "n_ratio",
-          "text"
-        ))
-        table$position <- 4
-        self$jaspResults[["powerEStab"]] <- table
-      } else {
-        return()
-      }
-
-      table$addColumnInfo(
-        name = "es",
-        title = gettext("True effect size"),
-        type = "string"
-      )
-      table$addColumnInfo(
-        name = "power",
-        title = gettext("Power to detect"),
-        type = "string"
-      )
-      table$addColumnInfo(
-        name = "desc",
-        title = gettext("Description"),
-        type = "string"
-      )
+.runTest1Var <- function(jaspResults, options) {
+  stats <- .prepareStats(options)
+
+  ## Compute results
+  results <- try(.computeTest1Var(jaspResults, options, stats))
+  if (inherits(results, "try-error")) {
+    .quitAnalysis(gettext("Unable to compute the power results. Try to enter less extreme values for the input parameters."))
+  }
```

Comment:
This seems to be on top of each of the analyses, could you move this into a function and just reuse the function?